docs(secrets): Add readme file for Git History (#4913)
bridgecrewio/checkov: docs/7.Scan Examples/Git History.md

---
layout: default
published: false
title: SCA scanning
nav_order: 20
---

# Git History

> Checkov supports scanning for secrets in git history, to identify and flag secrets which might not be in the current brunches, but are still visible by accessing the git history of the repo.

Reviewer: ```suggestion
Checkov supports scanning of secrets in git history to identify and flag secrets, which might not be in the current branches, but are still visible by accessing the git history of the repo.
```

Reviewer 2: ```suggestion
Checkov supports scanning for secrets in git history to identify and flag secrets that might not be in the head commit of the branch but are still visible by accessing the git history of the repo.
```

I want to avoid making it sound like we scan other branches.

Checkov fetches all the available commits, and uses the same strategy as in 'secrets' checkov framework to search for secrets in the diff available from git history.

> Each secret found, can be traced by the first commit in which it appeared, and the last commit, if it has been removed. 

Reviewer 2: ```suggestion
Each secret found can be traced to the first commit in which it appeared and the last commit that contained the secret if it was removed. 
```

## Git History scanning

> Git history scan is the same as secrets scan.

Reviewer 2: ```suggestion
Git history scan uses the same signatures and models as a regular secrets scan.
```

The only difference is that the root directory is the path to either the root git directory or the bare git repo.

> 

Reviewer 2: Need to add a line that say that `--scan-secrets-history` will only scan history for secrets and not other issues such as IaC misconfigs.

```suggestion
Using the `--scan-secrets-history` flag will scan git history for secrets only. This will not scan for other issues such as IaC misconfiguration. Use `--secrets-history-timeout` to set how long the secrets scan will run on history before stopping and returning results. The default is ___.
```

Reviewer 3: @tsmithv11 
since we changed it to run multi process we can't really return the results.
we run in parallel and if it's not finished properly we won't be able to compose the results correctly

Reviewer 2: Sounds good

```suggestion
Using the `--scan-secrets-history` flag will scan git history for secrets only. This will not scan for other issues such as IaC misconfiguration. Use `--secrets-history-timeout` to set how long the secrets scan will run on history before stopping and not returning results. The default is ___.
```

```bash

> checkov -d <git dir> --framework secrets --bc-api-key <your_api_key>

Reviewer: ```suggestion
checkov -d <git dir> --scan-secrets-history --bc-api-key <your_api_key>
```
also add an example with `--secrets-history-timeout` or an explanation on when to use this flag.
